lib/order.rb

```diff
@@ -1,18 +1,69 @@
+require 'csv'
+
 module Grocery
   class Order
     attr_reader :id, :products
-
+    # class variable for orders
     def initialize(id, products)
-      @id = id
+      @id = id.to_i
       @products = products
     end
+    # Class method that will allow up to return a collection of Order instances, represents orders described in CSV file
+    # array to hold the orders file information
+    def self.all
+      # Open this shit
+      file = CSV.open("support/orders.csv", "r")
+      # Loop through each line in CSV file and add it to the array orders
+      all_orders = []
+      file.each do |line|
+        # each line has fields separated by semi-colons, so split those fields. This tells us what the separaters are (';')
+        id = line[0]
+        raw_products = line[1]
+        products_1 = raw_products.split(";")
+        # puts "#{id}: #{raw_products}"
+        products_2 = []
+        products_1.each do |string|
+          products_2 << string.split(":")
+        end
+        final_products = {}
+        products_2.each do |product|
+          final_products[product[0]] = product[1]
+        end
+        all_orders << Order.new(id,final_products)
+      end
+      return all_orders
+    end
 
+    def self.find(id)
+      all_orders = Grocery::Order.all
+      all_orders.each do |order|
+        if all_orders.find(id) == id
+          return order
```

Comment:
I don't think this implementation of `find` will work. `all_orders` will be the array of orders you get back from `.all`. You start looping through it, which I like, but then you call `.find` on the whole list inside the loop.

Instead, you might do something like this:

```ruby
def self.find(id)
  all_orders = self.all
  all_orders.each do |order|
    if order.id == id
      return order
    end
  end
  return nil
end
```

What this method is doing is:

- Look at all the orders one by one
- If you find one with a matching ID, stop iterating and return it
- If you look at all the orders and haven't returned one yet, return nil

Reply:
Interestingly, you _did_ get almost exactly this implementation in `OnlineOrder`.